### PR TITLE
Update demo output; save image to file

### PR DESCRIPTION
**Pull request by vgoliber**
Add image to output for demos

### CODE REVIEW

File: tests/test_nurse_scheduling.py [hunk 1]
```diff
@@ -20,22 +20,20 @@
 
 class IntegrationTests(unittest.TestCase):
     @unittest.skipIf(os.getenv('SKIP_INT_TESTS'), "Skipping integration test.")
-    def test_scheduling(self):
+    def test_smoke(self):
+        """run nurse_scheduling.py and check that nothing crashes"""
+
+        demo_file = os.path.join(example_dir, 'nurse_scheduling.py')
+        subprocess.check_output([sys.executable, demo_file])
```
Comment (hhtong) [inline tests/test_nurse_scheduling.py]: Is there a reason for splitting the test into two and running the file twice?

File: nurse_scheduling.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 from dimod import BinaryQuadraticModel
 from collections import defaultdict
 from copy import deepcopy
+import matplotlib.pyplot as plt
```
Comment (hhtong) [inline nurse_scheduling.py]: Following the rest of the examples (until we drop py35):
```
try:
    import matplotlib.pyplot as plt
except ImportError:
    matplotlib.use("agg")
    import matplotlib.pyplot as plt
```

Comment (hhtong) [reply nurse_scheduling.py]: Also should add matplotlib to requirements.txt

File: nurse_scheduling.py [hunk 3]
```diff
@@ -163,52 +165,96 @@ def get_nurse_and_day(index):
...
-
-# min_duty_days sum
-sum_f = 0
-for n in range(n_nurses):
-    sum_d = 0
-    for d in range(n_days):
-        sum_d += preference * smpl[get_index(n, d)]
-    sum_f += lagrange_soft_nurse * (sum_d - min_duty_days) * (sum_d - min_duty_days)
-print("Checking Soft nurse constraint ", sum_f)
+# energy = results.first.energy
```
Comment (hhtong) [inline nurse_scheduling.py]: Can be deleted

**Comment by vgoliber** [comment]
@hhtong I have addressed all of your feedback.

**Review by hhtong — APPROVED** [review]

**Pull request merged**